### PR TITLE
CORE-18431 - Enable detekt scan for integration and smoke test code.

### PR DESCRIPTION
**Pull request by driessamyn**

### CODE REVIEW

**Review by ronanbrowne — APPROVED** [review]
LGTM (blt)

**Review by LWogan — APPROVED** [review]

**Review by lankydan — APPROVED** [review]
I've approve it, but honestly a lot of the formatting changes here look pretty bad and harder to read.

File: testing/p2p/inmemory-messaging-impl/src/integrationTest/kotlin/net/corda/messaging/emulation/subscription/stateandevent/InMemoryStateAndEventSubscriptionIntegrationTests.kt [hunk 1]
```diff
@@ -22,6 +18,10 @@ import org.junit.jupiter.api.Timeout
 import org.junit.jupiter.api.extension.ExtendWith
 import org.osgi.test.common.annotation.InjectService
 import org.osgi.test.junit5.service.ServiceExtension
+import java.util.*
```
Comment (nikinagy) [inline testing/p2p/inmemory-messaging-impl/src/integrationTest/kotlin/net/corda/messaging/emulation/subscription/stateandevent/InMemoryStateAndEventSubscriptionIntegrationTests.kt:21]: can we remove the * imports?

**Comment by driessamyn** [comment]
> I've approve it, but honestly a lot of the formatting changes here look pretty bad and harder to read.

They're the rulez ;) 

I was feed up with being told in PRs something doesn't comply to the rules because we didn't enforce them.
I don't mind what the rules are, as long as I don't need to reason about formatting things, one of those things a computer can do a much better job off.

File: testing/p2p/inmemory-messaging-impl/src/integrationTest/kotlin/net/corda/messaging/emulation/subscription/stateandevent/InMemoryStateAndEventSubscriptionIntegrationTests.kt (hunk 1 above)
Comment (driessamyn) [reply testing/p2p/inmemory-messaging-impl/src/integrationTest/kotlin/net/corda/messaging/emulation/subscription/stateandevent/InMemoryStateAndEventSubscriptionIntegrationTests.kt:21]: It seems like our detekt rules allow this one, so either we change the rule, or accept this.

**Review by kyriathar — APPROVED** [review]
OK (db & packaging)

File: testing/p2p/inmemory-messaging-impl/src/integrationTest/kotlin/net/corda/messaging/emulation/subscription/stateandevent/InMemoryStateAndEventSubscriptionIntegrationTests.kt (hunk 1 above)
Comment (nikinagy) [reply testing/p2p/inmemory-messaging-impl/src/integrationTest/kotlin/net/corda/messaging/emulation/subscription/stateandevent/InMemoryStateAndEventSubscriptionIntegrationTests.kt:21]: If we do that I am pretty sure there will be a lot more changes required.

**Review by nikinagy — APPROVED** [review]
LGTM (network)

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by nkovacsx — APPROVED** [review]

**Pull request merged**